### PR TITLE
#447

### LINKED ISSUE #447: Faucet - Provide control over the transaction chain and all nodes
### Is your feature request related to a problem?

Actually, if a user uses the faucet for an address, he couldn't receive more than 3 transactions in an hour with a node.
We should extend this control for the addresses of the transaction chain et for all nodes

### Describe the solution you'd like

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by prix-uniris**
# Description
This changes in following details:
- Make use of TransactionChain.fetch_genesis_address_remotely(address) in faucet rate limiter
- Triggering FaucetRateLimiter's register after transaction has been replicated and triggered from transaction_subscriber.ex

Fixes #447

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Manual testing with single node

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/archethic/transaction_chain.ex [hunk 1]
```diff
@@ -836,7 +836,7 @@ defmodule Archethic.TransactionChain do
     end
   end
 
-  defp fetch_genesis_address_remotely(address) when is_binary(address) do
+  def fetch_genesis_address_remotely(address) when is_binary(address) do
```
Comment (samuel-uniris) [inline lib/archethic/transaction_chain.ex:839]: Can you add some doc and typespec as well (as it's public now) ?

File: lib/archethic_web/faucet_rate_limiter.ex [hunk 2]
```diff
@@ -46,14 +48,23 @@ defmodule ArchethicWeb.FaucetRateLimiter do
...
   def handle_call({:block_status, address}, _from, state) do
+    address =
+      case TransactionChain.fetch_genesis_address_remotely(address) do
+        {:ok, genesis_address} -> genesis_address
+        _ -> address
+      end
+
     reply =
-      if address_state = Map.get(state, address) do
+      if address_state = Map.get(state, String.upcase(address)) do
```
Comment (samuel-uniris) [inline lib/archethic_web/faucet_rate_limiter.ex:67]: What is the purpose of `String.upcase` here ? 
Addresses should be of binary form

**Review by samuel-uniris — CHANGES_REQUESTED** [review]

File: lib/archethic_web/faucet_rate_limiter.ex (hunk 2 above)
Comment (prix-uniris) [reply lib/archethic_web/faucet_rate_limiter.ex:67]: This is just to store it in the map which is only the facuet_rate_limiter state. When user requests It comes as string so storing in facuet_rate_limiter as address -> some map values , where address is a string key and Upcase to keep consistency

File: lib/archethic/transaction_chain.ex (hunk 1 above)
Comment (prix-uniris) [reply lib/archethic/transaction_chain.ex:839]: Okay

File: lib/archethic_web/faucet_rate_limiter.ex (hunk 2 above)
Comment (samuel-uniris) [reply lib/archethic_web/faucet_rate_limiter.ex:67]: map keys can be binary, so the best would be to decode the input in the controller as binary and use it later, so you will remove the case issues

Comment (prix-uniris) [reply lib/archethic_web/faucet_rate_limiter.ex:67]: Okay makes sense.

**Review by samuel-uniris — APPROVED** [review]

**Pull request merged**